modules/controller/rpc-controller.js

```diff
@@ -137,83 +139,33 @@ class RpcController {
-            } else {
-                if(!this.isArrayOfStrings(req.body.assets)){
-                    return next({code: 400, message: `Assets must be a non-empty array of strings, all strings must have double quotes.`});
-                }
-                if(req.body.keywords && !this.isArrayOfStrings(req.body.keywords)) {
-                    return next({code: 400, message: `Keywords must be a non-empty array of strings, all strings must have double quotes.`});
-                }
-            }
-
-            const operationId = uuidv1();
-            try {
-                this.logger.emit({
-                    msg: 'Started measuring execution of publish command',
-                    Event_name: 'publish_start',
-                    Operation_name: 'publish',
-                    Id_operation: operationId
-                });
-
-                const inserted_object = await Models.handler_ids.create({
-                    status: 'PENDING',
-                });
-                const handlerId = inserted_object.dataValues.handler_id;
-                res.status(200).send({
-                    handler_id: handlerId,
-                });
-
-                const fileContent = req.files.file.data;
-                const fileExtension = path.extname(req.files.file.name).toLowerCase();
-                const assets = [...new Set(JSON.parse(req.body.assets.toLowerCase()))];
-                let visibility = JSON.parse(!!req.body.visibility);
-                let keywords = req.body.keywords ? JSON.parse(req.body.keywords.toLowerCase()) : [];
-
-                const assertion = await this.publishService.publish(fileContent, fileExtension, assets, keywords, visibility, handlerId);
-
-                const handlerData = {
-                    id: assertion.id,
-                    rootHash: assertion.rootHash,
-                    signature: assertion.signature,
-                    metadata: assertion.metadata,
-                };
-
-                await Models.handler_ids.update(
-                    {
-                        data: JSON.stringify(handlerData)
-                    }, {
-                        where: {
-                            handler_id: handlerId,
-                        },
-                    },
-                );
-
-            } catch (e) {
-                this.logger.error({
-                    msg: `Unexpected error at publish route: ${e.message}. ${e.stack}`,
-                    Event_name: constants.ERROR_TYPE.PUBLISH_ROUTE_ERROR,
-                    Event_value1: e.message,
-                    Id_operation: operationId
-                });
-            } finally {
-                this.logger.emit({
-                    msg: 'Finished measuring execution of publish command',
-                    Event_name: 'publish_end',
-                    Operation_name: 'publish',
-                    Id_operation: operationId
+        this.app.post('/api/latest/publish', async (req, res, next) => {
```

Comment:
This also breaks the dkg-client so we need to update it also

Reply:
Correct, after this PR I will start working on the API